dpypelines/s3_tar_received.py

```diff
@@ -1,22 +1,103 @@
+import importlib
+import json
+import os
+
+from dpytools.stores.directory.local import LocalDirectoryStore
+from dpypelines.pipeline.shared import notification as notify
+from dpypelines.pipeline.shared.schemas import get_config_schema_path
+from dpytools.s3.basic import decompress_s3_tar
+from dpytools.validation.json import validation
+from dpypelines.pipeline.shared import message
+
 def start(s3_object_name: str):
     """
     Handles the required behaviour when recieving a tar file indicated
     by an s3 object name.
 
-    Example s3_object_name: my-bucket/my-data.tar
+    Args:
+        s3_object_name (str): The S3 object name of the tar file to be processed.
+        Example: 'my-bucket/my-data.tar'
+
     """
 
-    # IN ALL CASES, the FOLLOWING STEPS WILL BE IN TRY CATCH
-    # BLOCKS AND WILL NOTIFY RELEVANT PARTIES IN EVENT OF FAILURE
+    # Decompress the tar file to the workspace
+    try:
+        decompress_s3_tar(s3_object_name, "input")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to decompress tar file {s3_object_name}", err))
+        raise err
+    
+    # Identify the child directory that was just decompressed within /inputs
+    try:
+        directories = [d for d in os.listdir("inputs") if os.path.isdir(os.path.join("inputs", d))]
+        assert len(directories) == 1, (
+            "Aborting, input directory has more than one directory within it"
+        )
+        decompressed_directory = directories[0]
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to identify decompressed directory in 'inputs'. Found directories: {directories}", err))
+        raise err
+
+    # Create a local directory store using the decompressed files
+    try:
+        localStore = LocalDirectoryStore(f"inputs/{decompressed_directory}")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to create local directory store at inputs/{decompressed_directory}", err))
+        raise err
+
+    # Verify if a pipeline configuration file ('pipeline-config.json') exists in the local directory store
+    try:
+        # Check for the existence of a configuration file
+        if not localStore.has_lone_file_matching("pipeline-config.json"):
+            msg = message.expected_local_file_missing("Missing pipeline-config.json", "pipeline-config.json", "start")
+            notify.data_engineering(msg)
+            # Create a default configuration file if one does not exist
```

Comment:
dont do this, I know there was a tbd (to be decided) but its still not decided. Just assume there is one and raise if there's not.

if we tackle this, it'll be as a separate piece later.

for now, just raise an exception here.